src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -402,6 +402,14 @@ public Intent getIntentForBrokerActivity(final AuthenticationRequest request) {
             if (intent != null) {
                 intent.putExtra(AuthenticationConstants.Broker.BROKER_REQUEST,
                         AuthenticationConstants.Broker.BROKER_REQUEST);
+                
+                // Only the new broker with PRT support can read the new PromptBehavior force_prompt. 
+                // If talking to the old broker, and PromptBehavior is set as force_prompt, reset it as 
+                // Always. 
+                if (!isBrokerWithPRTSupport(intent) && PromptBehavior.FORCE_PROMPT == request.getPrompt()) {
+                    Logger.v(TAG, "FORCE_PROMPT is set for broker auth via old version of broker app, reset to ALWAYS.");
+                    intent.putExtra(AuthenticationConstants.Broker.ACCOUNT_PROMPT, PromptBehavior.Always.name());
```

Comment:
old broker won't understand the force_prompt. so i checked the version here, if it's new broker, i pass whatever passed in to broker, if it's the old, i'll translate back to always.